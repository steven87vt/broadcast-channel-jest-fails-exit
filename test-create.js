import merge from 'lodash/merge'
import { createStore as createStoreImport } from './ClientApp/store'

export function createStore(overrides) {
  let http = {
    put: jest.fn(() => {
      Promise.resolve()
    }),
    post: jest.fn(() => {
      Promise.resolve()
    }),
    get: jest.fn(() => {
      Promise.resolve()
    }),
    delete: jest.fn(() => {
      Promise.resolve()
    })
  }

  let store = { plugins: [] }

  if (overrides && typeof overrides.http !== 'undefined' && typeof overrides.http === 'object') {
    merge(http, overrides.http)
  }
  if (overrides && typeof overrides.store !== 'undefined' && typeof overrides.store === 'object') {
    merge(store, overrides.store)
  }

  return createStoreImport(http, store)
}
