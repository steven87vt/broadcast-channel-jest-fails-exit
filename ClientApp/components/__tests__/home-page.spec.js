import Vuex from 'vuex'
import VueRouter from 'vue-router'

// import BroadcastChannel from 'broadcast-channel'

import { shallowMount, createLocalVue } from '@vue/test-utils'
import { createStore as createTestStore } from '../../../test-create'
import Router from '../../router'

import merge from 'lodash/merge'
import HomePage from '../home-page'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)

// const eventObject = {
//   $on: jest.fn(() => {
//     console.log("Event.$on triggered");
//     return Promise.resolve();
//   })
// };

function createWrapper(overrides) {
  const defaults = {
    localVue,
    router: Router,
    store: createTestStore()
  }

  return shallowMount(HomePage, merge(defaults, overrides))
}

describe('login.vue', () => {
  beforeEach(async () => {
    /*
     *let result = await BroadcastChannel.clearNodeFolder()s
     *console.log('Login.vue - BeforeEach - node directories removed: ' + result)
     */
  })

  test('login matches snapshot test.', () => {
    const wrapper = createWrapper()
    expect(wrapper).toMatchSnapshot()
    wrapper.destroy()
  })
})
