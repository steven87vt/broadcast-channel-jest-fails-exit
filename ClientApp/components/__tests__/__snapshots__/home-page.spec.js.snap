// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login.vue login matches snapshot test. 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
